subsys/net/openthread/rpc/server/ot_rpc_commissioning_server.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <nrf_rpc/nrf_rpc_cbkproxy.h>
+
+#include <zephyr/net/openthread.h>
+
+#include <openthread/link.h>
+
+static void ot_thread_discover_cb(otActiveScanResult *result, void *context, uint32_t callback_slot)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	struct ot_rpc_active_scan_result pkt_data;
+	size_t cbor_buffer_size = result ? sizeof(pkt_data) + 1 : 1;
+
+	cbor_buffer_size += 10; /* for context and callback slot */
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	if (result) {
+		memcpy(pkt_data.ext_addr, result->mExtAddress.m8, OT_EXT_ADDRESS_SIZE);
```

Comment:
I wonder if it's OK to serialize structures as BLOBs or we should encode each field as a separate CBOR data item. This is probably a topic for a broader discussion but we should take a unified strategy on this. I guess separate CBOR data items make it easier to validate the data correctness and detect client & server incompatibility when a structure format changes.

Reply:
BLOB was splitted for set of separate parameters. 1 cbor data per structure member.